opendcs-rest-api/src/test/java/org/opendcs/odcsapi/fixtures/TomcatServer.java

```diff
@@ -0,0 +1,152 @@
+{
+	private static final Logger logger = LoggerFactory.getLogger(TomcatServer.class);
+	private final SingleSignOnWrapper ssoValve = new SingleSignOnWrapper();
+	private final Tomcat tomcatInstance;
+	private final StandardContext context;
+	private Manager sessionManager;
+
+	/**
+	 * Setups the baseline for tomcat to run.
+	 *
+	 * @param baseDir     set to the CATALINA_BASE directory the build has set up
+	 * @param port        Network port to listen on
+	 * @param contextName url prefix to use, can be "/","/odcsapi" etc
+	 */
+	public TomcatServer(String baseDir, int port, String contextName) throws IOException
+	{
+
+		tomcatInstance = new Tomcat();
+		tomcatInstance.setBaseDir(baseDir);
+		Host host = tomcatInstance.getHost();
+
+
+		host.setAppBase("webapps");
+		String catalinaBase = tomcatInstance.getServer().getCatalinaBase().toString();
+		Files.createDirectories(Paths.get(catalinaBase, "temp"));
+		Files.createDirectories(Paths.get(catalinaBase, "webapps"));
+		tomcatInstance.setPort(port);
+		Connector connector = tomcatInstance.getConnector();
+		connector.setSecure(true);
+		connector.setScheme("https");
+		tomcatInstance.setSilent(false);
+		tomcatInstance.enableNaming();
+		Engine engine = tomcatInstance.getEngine();
+		logger.atTrace().log("Got engine " + engine.getDefaultHost());
+
+		host.addLifecycleListener(new HostConfig());
+		Pipeline pipeline = host.getPipeline();
+		pipeline.addValve(ssoValve);
+		context = (StandardContext) tomcatInstance.addWebapp(contextName, Paths.get("src/main/webapp").toAbsolutePath().toString());
+	}
+
+	public int getPort()
+	{
+		return tomcatInstance.getConnector().getLocalPort();
+	}
+
+	public Manager getTestSessionManager()
+	{
+		return this.sessionManager;
+	}
+
+	public SingleSignOnWrapper getSsoValve()
+	{
+		return this.ssoValve;
+	}
+
+	/**
+	 * Starts the instance of tomcat and returns when it's ready.
+	 *
+	 * @throws LifecycleException any error in the startup sequence
+	 */
+	public void start() throws LifecycleException
+	{
+		tomcatInstance.start();
+		this.sessionManager = context.getManager();
+		logger.atInfo().log("Tomcat listening at http://localhost:" + tomcatInstance.getConnector().getLocalPort());
+	}
+
+	/**
+	 * Used for the ./gradlew run command.
+	 * Unit tests only need to call start and move on.
+	 */
+	public void await()
+	{
+		tomcatInstance.getServer().await();
+	}
+
+	/**
+	 * Stops the instance of tomcat, including destroying the JNDI context.
+	 *
+	 * @throws LifecycleException any error in the stop sequence
+	 */
+	public void stop() throws LifecycleException
+	{
+		tomcatInstance.stop();
+	}
+
+	/**
+	 * arg[0] - the CATALINA_BASE directory you've setup
+	 * arg[1] - full path to the war file generated by this build script
+	 * arg[2] - name to use for this instance. See constructor for guidance
+	 *
+	 * @param args standard argument list
+	 */
+	public static void main(String[] args)
+	{
+		String baseDir = args[0];
+		String contextName = args[1];
```

Comment:
I saw that. :smile:. But considering this was lifted near verbatim from CDA I wonder if we should just make our little "embedded tomcat for test" it's own project.